Framework/Algorithms/src/FitPeaks.cpp

```diff
@@ -1069,11 +1069,12 @@ void FitPeaks::fitSpectrumPeaks(
     for (size_t i = 0; i < bkgdfunction->nParams(); ++i)
       bkgdfunction->setParameter(0, 0.);
 
-    // set the peak parameters from last good fit - override peak center
-    for (size_t i = 0; i < lastGoodPeakParameters.size(); ++i)
-      peakfunction->setParameter(i, lastGoodPeakParameters[i]);
     double expected_peak_pos = expected_peak_centers[peak_index];
     peakfunction->setCentre(expected_peak_pos);
+    // set matrix workspace so reads necessary params from .xml file after
+    // center set
+    peakfunction->setMatrixWorkspace(m_inputMatrixWS, wi, 0.0, 0.0);
+    peakfunction->clearTies(); // let S,A,B of B2B exp be refined
```

Comment:
I'm not sure you are preserving functionality here. Before this changeset the previous peak that was fit was used as the starting value for the next set. The purpose was that "near-ish" peaks in the same pixel tend to have the same basic shape. I haven't looked at the rest of the context to see if that still happens without the old 1072-1074

Reply:
This comment might be outdated, so sorry in advance if this has already been corrected. So my understanding was before my change the starting parameters to fit the next peak in the spectrum was initialised with those from the last peak fitted in that spectrum (the loop is over expected peaks in that spectrum). The centre was overwritten with the expected centre, so in principle it was just the height and width that was changed. Now we know the height of adjacent peaks at different d-spacing is not related (depends on the structure factor) and the width should be proportional to the d-spacing (TOF diffractometers have roughly constant delta d/d resolution at a given two theta) - i.e. it will be different too. Obviously for some peak shapes the width and height are estimated later so this is not a problem  - but that was not the case for the back to back exponential peaks. My change allows the last successful fit to a peak of the same nominal d-spacing in a different spectrum to be used as an initial guess. The chunking means that each thread loops over a contiguous set of workspace indices - which albeit are not forced to be at a similar position but often are - regardless the A,B and S will likely be similar as they are functions of the d-spacing (in GSAS). Obviously in the first loop when successful peak parameters haven't been found the peak starts from the default parameters - which is you have got a refinement in a parameters.xml file will be pretty good for the back to back exponentials (and the gaussian peaks have a width estimated anyway). The A,B ties for the back to back exponential are removed as of course not every instrument will have a Parameters.xml file and the starting values from the initialisation are not the best (I should really change them). Does the above sound reasonable?